extensions/container-image/container-image-s2i/deployment/src/main/java/io/quarkus/container/image/s2i/deployment/PackageUtil.java

```diff
@@ -0,0 +1,216 @@
+                        final boolean hasBasePath = !StringUtil.isBlank(base);
+                        final TarArchiveEntry entry = hasBasePath
+                                ? new TarArchiveEntry(base + File.separator + file.toFile())
+                                : new TarArchiveEntry(file.toFile());
+
+                        entry.setName(hasBasePath ? base + File.separator + relativePath.toString() : relativePath.toString());
+                        if (file.toFile().canExecute()) {
+                            entry.setMode(entry.getMode() | 0755);
+                        }
+                        entry.setSize(attrs.size());
+                        putTarEntry(tout, entry, file);
+                        return FileVisitResult.CONTINUE;
+                    }
+                });
+                tout.flush();
+            }
+            return tempFile;
+
+        } catch (IOException e) {
+            throw DekorateException.launderThrowable(e);
+        }
+    }
+
+    public static File packageFile(Path root, Path... additional) {
+        return packageFile(root, null, additional);
+    }
+
+    public static File packageFile(Path root, String base, Path... additional) {
+        try {
+            final Set<String> includes = Arrays
+                    .stream(additional)
+                    .map(p -> p.toAbsolutePath().toString())
+                    .collect(Collectors.toSet());
+
+            File tempFile = Files.createTempFile(Paths.get(DEFAULT_TEMP_DIR), DOCKER_PREFIX, BZIP2_SUFFIX).toFile();
+            try (final TarArchiveOutputStream tout = buildTarStream(tempFile)) {
+                Files.walkFileTree(root, new SimpleFileVisitor<Path>() {
+                    @Override
+                    public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) {
+                        return FileVisitResult.CONTINUE;
+                    }
+
+                    @Override
+                    public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                        String absolutePath = file.toAbsolutePath().toString();
+                        if (!shouldInclude(absolutePath, includes)) {
+                            return FileVisitResult.CONTINUE;
+                        }
+                        final Path relativePath = root.relativize(file);
+                        final boolean hasBasePath = !StringUtil.isBlank(base);
+                        final TarArchiveEntry entry = hasBasePath
+                                ? new TarArchiveEntry(base + File.separator + file.toFile())
+                                : new TarArchiveEntry(file.toFile());
+
+                        entry.setName(hasBasePath ? base + File.separator + relativePath.toString() : relativePath.toString());
+                        if (file.toFile().canExecute()) {
+                            entry.setMode(entry.getMode() | 0755);
```

Comment:
Are you sure that using `0755` like this is correct?

Reply:
Yeah